File: java/src/Filestore/WarehouseReader.java
package Filestore;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class WarehouseReader {

	public WarehouseReader(String filename) {
	}

	public List<StockMemento> readAll() {
		List<StockMemento> lines = new ArrayList<StockMemento>();
		lines.add(new StockMemento("01", "12", "1045", "3"));
		lines.add(new StockMemento("10", "04", "2761", "17"));
		lines.add(new StockMemento("24", "17", "5990", "50"));
		lines.add(new StockMemento("01", "01", "6000", "105"));
		return lines;
	}

}

File: dotnet/Filestore/WarehouseReader.cs
namespace Filestore {
	using System.Collections.Generic;

public class WarehouseReader {

	private string filename;

	public WarehouseReader(string filename) {
		this.filename = filename;
	}

	public IList<StockMemento> readAll() {
		List<StockMemento> lines = new ArrayList<StockMemento>();
		BufferedReader br = null;
		try {
			br = new BufferedReader(new FileReader(filename));
			string line;
			while ((line = br.readLine()) != null) {
				string[] args = line.Split(null);
				lines.Add(new StockMemento(args[0], args[1], args[2], args[3]));
			}
		} catch (IOException e) {
			throw new DataFileException(filename);
		} finally {
			try { if (br != null) br.close(); } catch (IOException ex) { }
		}
		return lines;
	}

}
}